src/common/cockpitwebresponse.c

```diff
@@ -1957,5 +1995,5 @@ cockpit_connection_get_protocol (GIOStream *connection,
          protocol = g_hash_table_lookup (headers, protocol_header);
     }
 
-  return protocol ? protocol : "http";
+  return protocol ?: (for_tls_proxy ? "https" : "http");
```

Comment:
Oh, tricky.  I think `?:` is a GCCism.  Are we fine with those?  I guess so.

Reply:
It seems the C99 standard isn't readily available for download.  I've seen this a lot, and we only care about gcc and clang, but this isn't important enough to quibble over -- I'll just revert it. But I'll wait for your full review, to avoid re-running everything twice.

Reply:
We actually already use it a lot -- `git grep '? *:'`

Reply:
Let's keep it, since we use it already a lot, as you have pointed out on IRC.